lib/dialects/abstract/query-generator.js

```diff
@@ -250,6 +250,9 @@ const QueryGenerator = {
       onDuplicateKeyUpdate += ' ON DUPLICATE KEY UPDATE ' + options.updateOnDuplicate.map(attr => {
         const field = rawAttributes && rawAttributes[attr] && rawAttributes[attr].field || attr;
         const key = this.quoteIdentifier(field);
+        if (options.preserveValuesOnNull && options.preserveValuesOnNull.indexOf(attr) !== -1) {
```

Comment:
Use dialect supports matrix, add an entry for `preserveOnNull` which should be true for mysql or any other supported dialects